js/blockchain.js

```diff
@@ -277,146 +306,185 @@ class MobileTwitter {
+            let jsonData = resp.data;
+            if (jsonData.hasOwnProperty('next_cursor_str') && String(jsonData.next_cursor_str) !== "0") {
+                this.cursor = jsonData.next_cursor_str;
+            } else {
+                this.have_ids_all_fetched = true;
+                console.log("INFO: all ids have been fetched");
+                // clear cursor when all ids fetched.
+                storage.setLocal({"block_cursor": {[currentProfileName]: null}},
+                    () => {
+                        console.log("INFO: " + "current cursor: " + this.cursor);
+                    });
+            }
+
             if (jsonData.hasOwnProperty('ids')) {
                 // split array of ids into 100 user chunks
                 let i = 0;
-                let chunks = [];
                 while (i * 100 < jsonData.ids.length) {
                     let slice = jsonData.ids.slice(i * 100, i * 100 + 100);
-                    chunks.push(slice.join(","));
+                    this.id_chunks.push(slice.join(","));
                     i++;
                 }
-                chunks = chunks.map((element) => {
-                    let url = 'users/lookup.json?include_entities=true&include_blocking=true'
-                    return this._makeRequest({
-                        url: url,
-                        headers: {
-                            'content-type': 'application/x-www-form-urlencoded'
-                        },
-                        body: `user_id=${element}`,
-                        method: 'POST'
-                    })
-                })
-                return Promise.all(chunks).then((values) => {
-                    values = values.map((users) => {
-                        return users;
-                    })
-                    return {
-                        users: values.flat(1)
-                    };
-                })
-            } else {
-                throw new Error('no ids');
             }
+        });
+    }
+
+    async* _getUserData() {
+
+        if (this.id_chunks.length === 0) {
+            finderRunning = false;
+            return Promise.resolve(false);
+        }
+
+        let rateLimitResetTime = Date.now();
+
+        while (this.id_chunks.length !== 0) {
+            let id_chunk = this.id_chunks.pop();
+            let url = 'users/lookup.json?include_entities=true&include_blocking=true';
+            yield this._makeRequest({
+                url: url,
+                headers: {
+                    'content-type': 'application/x-www-form-urlencoded'
+                },
+                body: `user_id=${id_chunk}`,
+                method: 'POST'
+            }).then(
+                (resp) => {
+                    rateLimitResetTime = this.obtainRateLimitResetTime(resp);
+                    return resp.data;
+                }
+            )
+            await sleep(rateLimitResetTime - Date.now() + 5000);
```

Comment:
Currently, the process is quite slow. I wonder it can be faster without getting blocked.

Reply:
Currently the extension is unusable on anyone with more than about 100 followers to block. I'd err on the side of making this slow and usable, and testing the api limit for faster times in a later PR.